packages/firestore/src/local/simple_db.ts

```diff
@@ -692,7 +715,35 @@ function wrapRequest<R>(request: IDBRequest): PersistencePromise<R> {
     };
 
     request.onerror = (event: Event) => {
-      reject((event.target as IDBRequest).error!);
+      const error = (event.target as IDBRequest).error!;
+      checkForAndReportiOSError(error);
+      reject(error);
     };
   });
 }
+
+// Guard so we only report the error once.
+let reportedIOSError = false;
+function checkForAndReportiOSError(error: DOMException): void {
+  if (reportedIOSError) {
+    return;
+  }
+  const iOSVersion = SimpleDb.getIOSVersion(getUA());
+  if (iOSVersion >= 12.2 && iOSVersion < 13) {
+    const IOS_ERROR =
+      'An internal error was encountered in the Indexed Database server';
+    if (error.message.indexOf(IOS_ERROR) >= 0) {
+      reportedIOSError = true;
+      // Throw a global exception outside of this promise chain, for the user to
+      // potentially catch.
+      setTimeout(() => {
```

Comment:
This could be `setImmediate`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate

> This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.

Reply:
Hm. I learned something today.

(Never develop JS for browsers).